src/libraries/Native/Unix/System.Globalization.Native/entrypoints.c

```diff
@@ -17,45 +18,69 @@ typedef uint16_t UChar;
 #include "pal_normalization.h"
 #include "pal_timeZoneInfo.h"
 
-#define FCFuncStart(name) EXTERN_C const void* name[]; const void* name[] = {
-#define FCFuncEnd() (void*)0x01 /* FCFuncFlag_EndOfArray */ };
-
-#define QCFuncElement(name,impl) \
-    (void*)0x8 /* FCFuncFlag_QCall */, (void*)(impl), (void*)name,
-
-FCFuncStart(gPalGlobalizationNative)
-    QCFuncElement("ChangeCase", GlobalizationNative_ChangeCase)
-    QCFuncElement("ChangeCaseInvariant", GlobalizationNative_ChangeCaseInvariant)
-    QCFuncElement("ChangeCaseTurkish", GlobalizationNative_ChangeCaseTurkish)
-    QCFuncElement("CloseSortHandle", GlobalizationNative_CloseSortHandle)
-    QCFuncElement("CompareString", GlobalizationNative_CompareString)
-    QCFuncElement("EndsWith", GlobalizationNative_EndsWith)
-    QCFuncElement("EnumCalendarInfo", GlobalizationNative_EnumCalendarInfo)
-    QCFuncElement("GetCalendarInfo", GlobalizationNative_GetCalendarInfo)
-    QCFuncElement("GetCalendars", GlobalizationNative_GetCalendars)
-    QCFuncElement("GetDefaultLocaleName", GlobalizationNative_GetDefaultLocaleName)
-    QCFuncElement("GetICUVersion", GlobalizationNative_GetICUVersion)
-    QCFuncElement("GetJapaneseEraStartDate", GlobalizationNative_GetJapaneseEraStartDate)
-    QCFuncElement("GetLatestJapaneseEra", GlobalizationNative_GetLatestJapaneseEra)
-    QCFuncElement("GetLocaleInfoGroupingSizes", GlobalizationNative_GetLocaleInfoGroupingSizes)
-    QCFuncElement("GetLocaleInfoInt", GlobalizationNative_GetLocaleInfoInt)
-    QCFuncElement("GetLocaleInfoString", GlobalizationNative_GetLocaleInfoString)
-    QCFuncElement("GetLocaleName", GlobalizationNative_GetLocaleName)
-    QCFuncElement("GetLocales", GlobalizationNative_GetLocales)
-    QCFuncElement("GetLocaleTimeFormat", GlobalizationNative_GetLocaleTimeFormat)
-    QCFuncElement("GetSortHandle", GlobalizationNative_GetSortHandle)
-    QCFuncElement("GetSortKey", GlobalizationNative_GetSortKey)
-    QCFuncElement("GetSortVersion", GlobalizationNative_GetSortVersion)
-    QCFuncElement("GetTimeZoneDisplayName", GlobalizationNative_GetTimeZoneDisplayName)
-    QCFuncElement("IndexOf", GlobalizationNative_IndexOf)
-    QCFuncElement("InitICUFunctions", GlobalizationNative_InitICUFunctions)
-    QCFuncElement("InitOrdinalCasingPage", GlobalizationNative_InitOrdinalCasingPage)
-    QCFuncElement("IsNormalized", GlobalizationNative_IsNormalized)
-    QCFuncElement("IsPredefinedLocale", GlobalizationNative_IsPredefinedLocale)
-    QCFuncElement("LastIndexOf", GlobalizationNative_LastIndexOf)
-    QCFuncElement("LoadICU", GlobalizationNative_LoadICU)
-    QCFuncElement("NormalizeString", GlobalizationNative_NormalizeString)
-    QCFuncElement("StartsWith", GlobalizationNative_StartsWith)
-    QCFuncElement("ToAscii", GlobalizationNative_ToAscii)
-    QCFuncElement("ToUnicode", GlobalizationNative_ToUnicode)
-FCFuncEnd()
+#ifndef lengthof
```

Comment:
It would be nice to extract these into a common header shared by all the System.* libraries instead of repeating the definitions of the macro and struct in each one. And maybe the lookup code too, creating a common function with (Entry*, count, name) arguments.